src/core/Form/TextInput/TextInput.baseStyles.tsx

```diff
@@ -44,6 +45,26 @@ export const baseStyles = withSuomifiTheme(
       box-shadow: ${theme.shadows.actionElementBoxShadow};
     }
     }
+    
+  &.fi-text-input_with-icon {
+    & .fi-text-input_container {
+    position: relative;
+    }
+
+    & .fi-text-input_input{
+    padding-right: ${math(
+      `${theme.spacing.insetXl} * 2 + ${theme.spacing.insetM}`,
```

Comment:
We should not use spacing tokens and inset tokens together, internal spacings inside components should use inset-tokens. This has probably been done elsewhere already, but I'd still avoid it.